src/App.jsx

```diff
@@ -14,33 +17,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD = {
+  name: 'name',
+  length: 'length',
+  none: 'none',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField, isReversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField !== SORT_FIELD.none) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD.name:
+          return good1.localeCompare(good2);
+        case SORT_FIELD.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SORT_FIELD.none);
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+  const shouldShowResetButton = sortField !== SORT_FIELD.none || isReversed;
```

Comment:
```suggestion
  const hasResetButton = sortField !== SORT_FIELD.none || isReversed;
```
Boolean variable should start from `is` or `has`